### PR TITLE
help people not familiarized with docker

### PR DESCRIPTION
**Pull request by facontidavide**
Not everyone use Docker to develop. I added some instruction for these folks. They may seem obvious to someone, but better to have them, than not

### CODE REVIEW

**Review by sea-bass — COMMENTED** [review]
This is great, but a few comments

1. Linting is failing, so look at that
2. I made a `media` folder to put images into, don't use the top folder
3. We are using `docker compose`, so you don't need to figure out the container name. For example

```
docker run <container name> bash
```

Becomes

```
docker compose run base
```

Where `base` is the name of the service we use.

Hopefully the `attach` commands you posted work, else `docker compose exec -it base bash` definitely works.

File: README.md [hunk 1]
```diff
@@ -102,3 +102,25 @@ It will ask you for your `sudo` password.
...
+docker attach <name of the container>
+```
+
+The name of the container will start with `roscon24-workshop-base-run-`.
+
+You can use auto-complete (tab) when attaching, or find the exact name using the command `docker ps`.
+
+### Develop inside the container using VSCode
+
+If you want to develop inside the container usinf VSCode, you need yo have the [Docker extension](https://marketplace.visualstudio.com/items?itemName=ms-azuretools.vscode-docker) installed.
```
Comment (sea-bass) [inline README.md:120]: usinf -> using

Comment (facontidavide) [reply README.md]: fixed

**Comment by facontidavide** [comment]
Thanks. I knew that there was a better way to attach... this demonstrates that there are people not familiar enough with docker (I am one of them)

File: .docker/entrypoint.sh [hunk 2]
```diff
@@ -23,7 +23,7 @@ alias delib_ws='cd $ROS_WS'
 function delib_build() {
         cwd=$(pwd)
         delib_ws
-        colcon build --symlink-install --continue-on-error
+        colcon build --symlink-install --continue-on-error --mixin compile-commands
```
Comment (sea-bass) [inline .docker/entrypoint.sh:26]: If we're including this mixin change, include it in the other bash functions that also call `colcon build`

File: README.md [hunk 3]
```diff
@@ -70,12 +70,20 @@ Note that this will pull the Docker image and may take a few minutes.
...
 Once you're in the container, check that you can launch a simulated world and the PyRoboSim UI appears on your screen.
 
 ```bash
 ros2 run delib_ws_worlds run
 ```
 
+You can attach to the running container from a different console, using the command:
+
+```bash
+docker compose run base bash
```
Comment (sea-bass) [inline README.md:84]: Did this work for you, or did you need to add `run -it base bash` for interactive / tty? (That's the usual way to attach)

**Review by sea-bass — APPROVED** [review]
Only some small stuff, :shipit:

File: README.md (hunk 3 above)
Comment (dcconner) [reply README.md:84]: `docker compose run base` seems to work for me.  Are we defaulting to `-it` in the entry point?

Simpler is better for our workshop commands.

File: .docker/entrypoint.sh (hunk 2 above)
Comment (facontidavide) [reply .docker/entrypoint.sh:26]: ok

File: README.md (hunk 3 above)
Comment (facontidavide) [reply README.md:84]: it worked without the `-it`

**Comment by facontidavide** [comment]
I believe that I have addressed all the concerns / questions. Thanks!!!! i am merging it

**Pull request merged**